lib/features/thread/data/repository/thread_repository_impl.dart

```diff
@@ -176,32 +176,40 @@ class ThreadRepositoryImpl extends ThreadRepository {
     Properties? updatedProperties,
     List<Email>? emailCacheList
   }) async {
-    if (emailUpdated != null && emailUpdated.isNotEmpty) {
-      if (updatedProperties == null) {
-        return null;
-      }
-      final newEmailUpdated = emailUpdated
-        .map((updatedEmail) => _combineUpdatedWithEmailInCache(updatedEmail, emailCacheList))
-        .where((tuple) => tuple.value2 != null)
-        .map((tuple) => tuple.value2!.combineEmail(tuple.value1, updatedProperties))
-        .toList();
+    if (emailUpdated == null || emailUpdated.isEmpty) return emailUpdated;
```

Comment:
should write test to make sure this method work well

Reply:
Add